fix: bspConfig sbt.script - use `Path` env variable as a fallback for Windows

dos65 opened the pull request:
Originally reported in https://github.com/scalameta/metals/issues/4702#issuecomment-1339137772

lightbend-cla-validator commented:
At least one commit author (v.checlyshov@tinkoff.ru) is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

dos65 committed:
fix: bspConfig sbt.script - use `Path` env variable as a fallback for Windows

Originally reported in https://github.com/scalameta/metals/issues/4702#issuecomment-1339137772

eed3si9n commented:
@dos65 Thanks for the contribution!

eed3si9n commented on protocol/src/main/scala/sbt/internal/bsp/BuildServerConnection.scala:
```diff
@@ -63,7 +63,7 @@ object BuildServerConnection {
     // As a fallback we try to find the sbt script in $PATH
     val fileName = if (Properties.isWin) "sbt.bat" else "sbt"
-    val envPath = sys.env.getOrElse("PATH", "")
+    val envPath = sys.env.get("PATH").orElse(sys.env.get("Path")).getOrElse("")
```
Instead of singling out `PATH` and `Path`, would it make sense at this point to filter to all keys that becomes `PATH` if you turned it to upper case, and pick the first entry?

dos65 replied:
Yep, it makes sense. Changed

martingd replied:
@eed3si9n Doesn't it potentially break semantics on *NIX where you could have a variable PATH and also other case variants?
The suggested code gives no guarantee which would be the first.

I would suggest that the “filter to all keys that becomes PATH if you turned it to upper case, and pick the first entry” was only used on Windows (to match Windows semantics) but on *NIX you still only looked for PATH (upper case) We already have special handling for Windows for the name of the sbt launcher.

dos65 replied:
@martingd 
> We already have special handling for Windows for the name of the sbt launcher

Could you point where is this handling defined?
I think the best option would be to reuse/copy-paste the existing logic
